packages/wallet/test/integration/withdrawal.test.ts

```diff
@@ -1,23 +1,76 @@
 import { Cardano } from '@cardano-sdk/core';
-import { SingleAddressWallet, TransactionFailure } from '../../src';
+import { firstValueFrom, of } from 'rxjs';
+import { logger } from '@cardano-sdk/util-dev';
+
+import * as mocks from '../mocks';
+import { RewardAccount, SingleAddressWallet, StakeKeyStatus, TransactionFailure, buildTx } from '../../src';
+import { assertTxIsValid } from '../util';
 import { createWallet } from './util';
-import { firstValueFrom } from 'rxjs';
 
 describe('integration/withdrawal', () => {
   let wallet: SingleAddressWallet;
+  let rewardAccounts: RewardAccount[];
 
-  beforeAll(async () => {
+  beforeEach(async () => {
     ({ wallet } = await createWallet());
+    rewardAccounts = [
+      {
+        address: Cardano.RewardAccount('stake_test1uqu7qkgf00zwqupzqfzdq87dahwntcznklhp3x30t3ukz6gswungn'),
+        delegatee: {
+          currentEpoch: undefined,
+          nextEpoch: undefined,
+          nextNextEpoch: undefined
+        },
+        keyStatus: StakeKeyStatus.Registered,
+        rewardBalance: 33_333n
+      },
+      {
+        address: Cardano.RewardAccount('stake1u89sasnfyjtmgk8ydqfv3fdl52f36x3djedfnzfc9rkgzrcss5vgr'),
+        delegatee: {
+          currentEpoch: undefined,
+          nextEpoch: undefined,
+          nextNextEpoch: undefined
+        },
+        keyStatus: StakeKeyStatus.Unregistered,
+        rewardBalance: 44_444n
+      }
+    ];
   });
 
   it('has balance', async () => {
     expect(typeof (await firstValueFrom(wallet.balance.utxo.total$))?.coins).toBe('bigint');
     expect(typeof (await firstValueFrom(wallet.balance.rewardAccounts.rewards$))).toBe('bigint');
   });
 
+  it('does not withdraw from accounts without rewards', async () => {
```

Comment:
```suggestion
  it('does not withdraw from reward accounts with zero balance when there are others with a non-zero balance', async () => {
```
Please also add a test to cover this assertion:

```ts
expect(tx.body.withdrawals).toBeUndefined();
```

Reply:
It is an empty array when there are no rewards. Should it be undefined?

Reply:
Verified 0 length withdrawals array: [4597374](https://github.com/input-output-hk/cardano-js-sdk/pull/446/commits/45973749aec4661245b30de06b49fb73c6a8f997)

Reply:
`undefined` is the expected assertion though, as it's a nullable property.

Reply:
[5e820cc](https://github.com/input-output-hk/cardano-js-sdk/pull/446/commits/5e820ccc80bfffe6111f9b2bef0ba2117be74935)
- Undefined `withdrawals` when rewards have 0 balance